examples/vnet/gw.c

```diff
@@ -127,19 +127,19 @@ static struct doca_flow_fwd_tbl *gw_build_port_fwd(int port_id)
 }
 
 /*RSS not include core0, for n_queue: 1 - n_queue*/
-static struct doca_flow_fwd_tbl *gw_build_rss_fwd(int n_queues)
+static struct doca_flow_fwd_tbl *gw_build_rss_fwd(struct gw_rss_cfg *rss_cfg)
 {
-	int i;
+	uint32_t i;
 	struct doca_flow_fwd cfg = {0};
 	uint16_t *queues;
 
-	queues = malloc(sizeof(uint16_t) * n_queues);
-	for (i = 1; i < n_queues; i++)
-		queues[i - 1] = i;
+	queues = malloc(sizeof(uint16_t) * rss_cfg->num_queues);
+	for (i = 0; i < rss_cfg->num_queues; i++)
+		queues[i] = rss_cfg->queue[i];
 	cfg.type = DOCA_FWD_RSS;
 	cfg.rss.queues = queues;
-	cfg.rss.rss_flags = DOCA_RSS_IP | DOCA_RSS_UDP | DOCA_RSS_IP;
-	cfg.rss.num_queues = n_queues - 1;
+	cfg.rss.rss_flags = rss_cfg->rss_flags;
+	cfg.rss.num_queues = rss_cfg->num_queues;
```

Comment:
user will not use RSS cfg to do hairpin just because hairpin is working on queues currently.
We have dedicated FWD struct and if user puts port there and we are in nic mode we will do the hairpin

struct doca_flow_fwd {
    enum doca_flow_fwd_type type;
    union {
        struct fwd_rss {
                uint32_t rss_flags;
                uint16_t *queues;
                int num_queues;
        } rss;

        struct port {
                int id;
        } port;

        struct next_pipelne {
            struct doca_flow_pipeline *next;
        } next_pipeline;
    };
};

Reply:
fix it, with new commits